### DISCUSSION TITLE
"The redraw model/supports are not behaving as I expected"

### DISCUSSION THREAD

**Discussion by ssotabe** [sn4k3/UVtools#926] [Q&A]
Hi,
I want to use the redraw model/supports to set all the supports to a brightness of 255 (as some parts have become darker due to other operations). However, even when using support mode, it can only darken the current pixel brightness, not increase it.

Is there any way to fill only the supports with a brightness of 255? Any advice would be greatly appreciated.

**Comment by sn4k3** [comment]
The tool let you choose the brightness, you can put 255 to draw at full white. Make sure to have the model B sliced without supports enabled.

**Reply by ssotabe** [reply]
I can darken the existing values (e.g., from 200 to 150), but I can't make them brighter (e.g., from 200 to 255). Additionally, no history is generated.
![image](https://github.com/user-attachments/assets/fba9d62a-758d-45e8-95ec-fcd55d67e790)
![image](https://github.com/user-attachments/assets/dff863e1-4f63-432e-bf8c-499673d12b56)
↓
![image](https://github.com/user-attachments/assets/9608ee5d-a768-4990-b261-5e811427d8ef)
↓
![image](https://github.com/user-attachments/assets/38ab2053-ed23-41ee-9b92-2498b317a9b8)
![image](https://github.com/user-attachments/assets/2e49ccae-f5dd-46f5-b043-dd25d70da49d)

**Reply by sn4k3** [reply]
Can you send me your model A + B?

**Reply by ssotabe** [reply]
[slice.zip](https://github.com/user-attachments/files/16981930/slice.zip)
It is different from the model above, but it was reproducible.

**Reply by sn4k3** [reply]
The redraw operation can only subtract pixel brightness, the Brightness field will do the following "pixel = pixel - (255 - brightness input)".
In your case it will end in subtract 0, (255 - 255) = 0, as so brightness is unaffected. 
If this is important I can add a operator to let user choose from Subtract, Add, Set or Max.

**Reply by ssotabe** [reply]
Thank you for the explanation. I understand that the current redraw operation only reduces brightness. Depending on the use case, other operations might be necessary, so if possible, it would be helpful to have an option for users to choose from subtraction, addition, setting, or maximum value.

**Reply by sn4k3** [reply]
Try new version

**Reply by ssotabe** [reply]
Thank you for your response.
I have tried the SET but it does not appear to be working as expected.
When Model(B) is SET for Model+support(A), only the pixels contained in Model(B) are SET, and Support disappears.

**Reply by sn4k3** [reply]
As I understand you are now using model B as the full model with supports (Body field)? That way won't work, the tool was made where model B must be the model A (full) sliced without supports, raft and flush to bed without elevation.
Inverting A with B will cause trouble as tool is not inteded to be used like that. The tooltip on the field indicates that:

![image](https://github.com/user-attachments/assets/d0f768c4-81cc-43f4-b2bc-fcc3352c17f2)
